lua/telescope/utils.lua

```diff
@@ -361,6 +364,11 @@ utils.transform_path = function(opts, path)
       if vim.tbl_contains(path_display, "shorten") or path_display["shorten"] ~= nil then
         transformed_path = Path:new(transformed_path):shorten(path_display["shorten"])
       end
+      if vim.tbl_contains(path_display, "truncate") or path_display.truncate then
+        local status = get_status(vim.api.nvim_get_current_buf())
+        local width = vim.api.nvim_win_get_width(status.results_win) - status.picker.selection_caret:len() - 2
```

Comment:
We could hide/cache the width in status or opts, depending how long it takes to get the status. prefixed with `__` or so.

We could benchmark it ;)

Reply:
Couldn't figure out an easy way to benchmark it since there are lots of moving pieces.
Have stored the length in `opts` as `__length`.